yabt/builders/golang.py

```diff
@@ -82,24 +84,140 @@ def go_prog_manipulate_target(build_context, target):
 
 @register_build_func('GoProg')
 def go_prog_builder(build_context, target):
-    """Build a Go binary executable"""
-    yprint(build_context.conf, 'Build GoProg', target)
-    workspace_dir = build_context.get_workspace('GoProg', target.name)
-    rmtree(workspace_dir)
+    go_prog_builder_internal(build_context, target, command='build')
+
+
+register_builder_sig(
+    'GoTest',
+    [('sources', PT.FileList),
+     ('in_buildenv', PT.Target),
+     ('cmd_env', None),
+     ('go_package', PT.str, None),
+     ('mod_file', PT.File, None),
+     ])
+
+
+@register_manipulate_target_hook('GoTest')
+def go_prog_manipulate_target(build_context, target):
+    target.buildenv = target.props.in_buildenv
+
+
+@register_build_func('GoTest')
+def go_prog_builder(build_context, target):
+    target.tags.add('testable')
+    go_prog_builder_internal(build_context, target, command='test')
+
+
+def rm_all_but_go_mod(workspace_dir):
```

Comment:
1. I don't like this
2. I don't understand the relation between a `go.mod` for a specific Prog/Test and the global `yabtwork/go` dir that stores downloaded packages. how does that work with multiple Go applications in the same project? by definition, to `go.mod` file will be different for each one, by they all use the same packages download dir? this feels risky and non-hermetic. what did we do about node_modules - it feels similar.